src/main/scala/org/ergoplatform/settings/Algos.scala

```diff
@@ -38,4 +38,26 @@ object Algos extends ScorexEncoding {
   def merkleTreeRoot(elements: Seq[LeafData]): Digest32 =
     if (elements.isEmpty) emptyMerkleTreeRoot else MerkleTree(elements)(hash).rootHash
 
+  /**
+    * Add longs, returning Long.Max value if there was any long overflow
+    */
+  @inline def addExact(a: Long, b: Long): Long = {
+    val sum = a + b
+    if (sum < 0) Long.MaxValue else sum
```

Comment:
What if I pass negative arguments? It not obvious that the output will be `Long.MaxValue`